### PR TITLE
[CARBONDATA-2262] Support the syntax of 'using CARBONDATA' to create table

### PR DESCRIPTION
**Pull request by xubo245**
Add new function to Support the syntax of 'using CARBONDATA' to create table, for example:

`CREATE TABLE src_carbondata1(key INT, value STRING) using carbondata `
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 
 - [x] Any backward compatibility impacted?
 
 - [x] Document update required?

 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4443/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3210/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3965/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3217/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4450/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3969/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3971/

**Comment by xubo245** [comment]
@jackylk Please review it.

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/sql/commands/UsingCarbondataSuite.scala [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  override def afterEach(): Unit = {
+    sql("DROP TABLE IF EXISTS src_carbondata1")
+    sql("DROP TABLE IF EXISTS tableSize3")
+  }
+
+  test("CARBONDATA-2262: test check results of table with complex data type and bucketing") {
+    sql("drop table if exists create_source")
+    sql("create table create_source(intField int, stringField string, complexField array<int>) " +
+      "USING CARBONDATA OPTIONS('tableName'='create_source')")
```
Comment (jackylk) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/sql/commands/UsingCarbondataSuite.scala]: no need to give tableName option, please modify

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/DDLStrategy.scala [hunk 2]
```diff
@@ -186,15 +186,17 @@ class DDLStrategy(sparkSession: SparkSession) extends SparkStrategy {
         ExecutedCommandExec(CarbonResetCommand()) :: Nil
       case org.apache.spark.sql.execution.datasources.CreateTable(tableDesc, mode, None)
         if tableDesc.provider.get != DDLUtils.HIVE_PROVIDER
-           && tableDesc.provider.get.equals("org.apache.spark.sql.CarbonSource") =>
+          && (tableDesc.provider.get.equals("org.apache.spark.sql.CarbonSource")
+          || tableDesc.provider.get.equalsIgnoreCase("CARBONDATA")) =>
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/DDLStrategy.scala]: make it non capital

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/commands/SparkSessionForUsingCarbonDataExample.scala [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+import java.io.File
+
+import org.apache.commons.io.FileUtils
+import org.apache.spark.sql.SparkSession
+
+import org.apache.carbondata.core.constants.CarbonCommonConstants
+import org.apache.carbondata.core.util.CarbonProperties
+
+object SparkSessionForUsingCarbonDataExample {
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/commands/SparkSessionForUsingCarbonDataExample.scala]: Don't suggest adding a new example, can add the new usage into the current example:SparkSessionExample

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/sql/commands/UsingCarbondataSuite.scala (hunk 1 above)
Comment (xubo245) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/sql/commands/UsingCarbondataSuite.scala]: ok, done

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/DDLStrategy.scala (hunk 2 above)
Comment (xubo245) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/strategy/DDLStrategy.scala]: ok, done

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/commands/SparkSessionForUsingCarbonDataExample.scala (hunk 3 above)
Comment (xubo245) [reply examples/spark2/src/main/scala/org/apache/carbondata/examples/commands/SparkSessionForUsingCarbonDataExample.scala]: OK, done

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3394/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4619/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4120/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4122/

**Comment by xubo245** [comment]
retest this please

**Comment by xubo245** [comment]
retest sdv please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3399/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4624/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4128/

**Comment by xubo245** [comment]
retest sdv please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4129/

**Comment by xubo245** [comment]
retest sdv please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4136/

**Comment by xubo245** [comment]
retest sdv please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4144/

**Comment by xubo245** [comment]
@jackylk @chenliang613 CI pass, Please check it.

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/SparkSessionExample.scala [hunk 4]
```diff
@@ -63,123 +63,126 @@ object SparkSessionExample {
...
-         | stringField STRING,
-         | timestampField TIMESTAMP,
-         | decimalField DECIMAL(18,2),
-         | dateField DATE,
-         | charField CHAR(5)
-         | )
-         | USING org.apache.spark.sql.CarbonSource
-         | OPTIONS('DICTIONARY_INCLUDE'='dateField, charField',
-         | 'dbName'='default', 'tableName'='sparksession_table')
+    for (keyWord <- Array("org.apache.spark.sql.CarbonSource", "carbondata")) {
```
Comment (jackylk) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/SparkSessionExample.scala]: I think this makes it more complex, just use `carbondata` is enough

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4651/

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3424/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4654/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3427/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4161/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4162/

**Pull request closed without merging**